src/tree_model_filter.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2013-2015, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use ffi;
+use glib::object::{Downcast, IsA};
+use glib::translate::*;
+use std::mem::transmute;
+
+use glib_ffi::{gboolean, gpointer};
+use {TreeIter, TreeModel, TreeModelFilter};
+use ffi::{GtkTreeIter, GtkTreeModel};
+
+unsafe extern "C" fn trampoline<T>(this: *mut GtkTreeModel, iter: *mut GtkTreeIter,
```

Comment:
Since we don't have a `TreeModelFilterExt` trait here there's no point making these functions generic.